### DISCUSSION TITLE
A better way of a singular model containing multiple models?

### DISCUSSION THREAD

**Discussion by piperun** [prisma/prisma#20248] [Q&A] [answered]
I do apologize if this is not the right place to ask these kinds of questions, but I couldn't find an official Prisma discord/slack.
Right now I've got a model that effectively looks like this:
```
model Shelf {
  books Book[]
  images Images[]
  childShelves Shelf[] @relation("ParentToChild")
  parentShelf Shelf?  @relation("ParentToChild", fields: [parentId], references: [id])
  parentId     String?
}
```

However I've felt this model isn't the most flexible and I've looked into a model that looks something like this:
```
model Shelf {
  items Item[]
}
model Item {
  books Book[]
  images Images[]
  shelves Shelf[]
}
```

Now this does not translate 1:1, since the only two models I've manage to come up with that is what I want are:
```
model Shelf {
  items Item[]
  childShelves Shelf[] @relation("ParentToChild")
  parentShelf Shelf?  @relation("ParentToChild", fields: [parentId], references: [id])
  parentId     String?
}
```
or
```
model Shelf {
  items Item[] @relation("ShelfToItem")
  itemShelf  Item[] @relation("ItemToShelf")
}

model Item {
  shelves     Shelf[] @relation("ShelfToItem")
  childShelf Shelf?  @relation("ItemToShelf", fields: [shelfId], references: [id])
  sheflId   Int?
}
```

**Comment by ludralph** [comment, marked as answer]
Hi @piperun 👋 

Based on the information you provided, it seems that your main requirement is to have a data model that allows a singular `Shelf` entity to contain multiple `Item` entities, where each `Item` can have multiple `Books` and `Images`. Additionally, you want to support a hierarchical relationship between shelves, where a `Shelf` can have child shelves and a reference to a parent shelf. Considering the requirements which I've specified and would need more clarification if it doesn't meet your needs, I would suggest this data model:

```prisma
model Shelf {
  id          Int       @id @default(autoincrement())
  items       Item[]
  childShelves Shelf[]   @relation("ParentToChild")
  parentShelf Shelf?     @relation("ParentToChild", fields: [parentId], references: [id])
  parentId    Int?
}

model Item {
  id      Int     @id @default(autoincrement())
  books   Book[]
  images  Images[]
  shelves Shelf[]
}
```
Note that you would need to also create the actual `Book` and `Images` as well.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by piperun** [reply]
Sorry for my late reply, but you are correct!
